eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java

```diff
@@ -43,7 +42,13 @@ private SetIterables()
             SetIterable<? extends E> setA,
             SetIterable<? extends E> setB)
     {
-        return SetIterables.unionInto(setA, setB, UnifiedSet.newSet());
+        if (setA.size() > setB.size())
+        {
+            UnifiedSet<E> result = UnifiedSet.newSet(setA);
```

Comment:
The following code should reduce bytecode size and increase inlining chance:
```java
if (setA.size() > setB.size())
{
    SetIterable<? extends E> tmp = setA;
    setA = setB;
    setB = tmp;
}
```
(and below)